sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -186,6 +186,22 @@ static void *_event_loop(void *args)
     /* register interest in all IPv6 packets */
     gnrc_netreg_register(GNRC_NETTYPE_IPV6, &me_reg);
 
+    DEBUG_PUTS("ipv6: wait for interfaces to be ready");
+    while (gnrc_netif_iter(NULL) == NULL) {
+        ztimer_sleep(ZTIMER_MSEC, 250);
+    }
```

Comment:
This looks fragile. But I also do not know any way to reliably wait for interface registration to complete off the top of my head or even if such feature exists :/

Reply:
The interfaces could send a message when they are initialized, then we can handle the registration here.